test_conformance/subgroups/test_workgroup.cpp

```diff
@@ -1651,38 +1652,43 @@ struct RED_NU_LG {
 
 // DESCRIPTION:
 // Test for reduce cluster functions
-// Which: 0 - add, 1 - max, 2 - min, 3 - mul, 4 - and, 5 - or, 6 - xor
+// Which: 0 - add, 1 - max, 2 - min, 3 - mul, 4 - and, 5 - or, 6 - xor, 7 - logical and, 8 - logical or, 9 - logical xor
+template <typename Ty, int Which>
+struct RED_OP;
```

Comment:
Could we make this struct more generic? It could be reused for other tests, like tests for cl_khr_subgroup_non_uniform_arithmetic. We could name it just OP or OPERATION and it would be useful to properly handle: https://github.com/aratajew/OpenCL-CTS/pull/4

Reply:
Sure, I'll do it.

Reply:
Done in a0d3544